climada/hazard/trop_cyclone.py

```diff
@@ -387,38 +387,69 @@ def _tc_from_track(self, track, centroids, coastal_idx, model='H08',
         new_haz.basin = [track.basin]
         return new_haz
 
-    def _apply_criterion(self, criterion, scale):
+    def _apply_knutson_criterion(self, chg_int_freq, scaling_rcp_year):
         """Apply changes defined in criterion with a given scale
         Parameters:
             criterion (list(dict)): list of criteria
             scale (float): scale parameter because of chosen year and RCP
         Returns:
             TropCyclone
         """
-        haz_cc = copy.deepcopy(self)
-        for chg in criterion:
-            # filter criteria
-            select = np.ones(haz_cc.size, bool)
-            for var_name, cri_val in chg['criteria'].items():
-                var_val = getattr(haz_cc, var_name)
-                if isinstance(var_val, list):
-                    var_val = np.array(var_val)
-                tmp_select = np.logical_or.reduce([var_val == val for val in cri_val])
-                select = select & tmp_select
-            if chg['function'] == np.multiply:
-                change = 1 + (chg['change'] - 1) * scale
-            elif chg['function'] == np.add:
-                change = chg['change'] * scale
-            if select.any():
-                new_val = getattr(haz_cc, chg['variable'])
-                # 1d-masks like `select` are inefficient for indexing sparse matrices since
-                # they are broadcasted densely in the second dimension
-                if isinstance(new_val, sparse.csr_matrix):
-                    new_val = sparse.diags(np.where(select, change, 1)).dot(new_val)
-                else:
-                    new_val[select] *= change
-                setattr(haz_cc, chg['variable'], new_val)
-        return haz_cc
+        tc_cc = copy.deepcopy(self)
+        tc_cc_basin = np.array(tc_cc.basin)
+        
+        #criterion per basin
```

Comment:
1. Remove trailing white space (https://www.python.org/dev/peps/pep-0008/#other-recommendations)
2. https://www.python.org/dev/peps/pep-0008/#block-comments
```suggestion

        # criterion per basin
```